src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/VariableFetchAnalyzer.php

```diff
@@ -531,26 +540,231 @@ public static function getGlobalType(string $var_id): Union
         }
 
         if ($var_id === '$argv') {
+            // only in CLI, null otherwise
             return new Union([
-                new TArray([Type::getInt(), Type::getString()]),
+                new TNonEmptyList(Type::getString()),
+                new TNull()
             ]);
         }
 
         if ($var_id === '$argc') {
-            return Type::getInt();
+            // only in CLI, null otherwise
+            return new Union([
+                new TIntRange(1, null),
+                new TNull()
+            ]);
+        }
+
+        if (!self::isSuperGlobal($var_id)) {
+            return Type::getMixed();
         }
 
         if ($var_id === '$http_response_header') {
             return new Union([
-                new TList(Type::getString())
+                new TList(Type::getNonEmptyString())
+            ]);
+        }
+
+        if ($var_id === '$GLOBALS') {
+            return new Union([
+                new TNonEmptyArray([
+                    Type::getNonEmptyString(),
+                    Type::getMixed()
+                ])
+            ]);
+        }
+
+        if ($var_id === '$_COOKIE') {
+            $type = new TArray(
+                [
+                    Type::getNonEmptyString(),
+                    Type::getString(),
+                ]
+            );
+
+            return new Union([$type]);
+        }
+
+        if (in_array($var_id, array('$_GET', '$_POST', '$_REQUEST'), true)) {
+            $array_key = new Union([new TNonEmptyString(), new TInt()]);
+            $array = new TNonEmptyArray(
+                [
+                    $array_key,
+                    new Union([
+                        new TString(),
+                        new TArray([
+                            $array_key,
+                            Type::getMixed()
+                        ])
+                    ])
+                ]
+            );
+
+            $type = new TArray(
+                [
+                    $array_key,
+                    new Union([new TString(), $array]),
+                ]
+            );
+
+            return new Union([$type]);
+        }
+
+        if ($var_id === '$_SERVER' || $var_id === '$_ENV') {
+
+            $string_helper = Type::getNonEmptyString();
+            $string_helper->possibly_undefined = true;
+
+            $non_empty_string_helper = Type::getNonEmptyString();
+            $non_empty_string_helper->possibly_undefined = true;
+
+            $argv_helper = new Union([
+                new TNonEmptyList(Type::getString())
+            ]);
+            $argv_helper->possibly_undefined = true;
+
+            $argc_helper = new Union([
+                new TIntRange(1, null)
+            ]);
+            $argc_helper->possibly_undefined = true;
+
+            $request_time_helper = new Union([
+                new TIntRange(time(), null)
```

Comment:
I get the meaning here but I'm afraid users may find confusing the fact that they retrieve a seemingly random value like `1663105964` here, WDYT?

Reply:
Alright :)